src/pages/campaign/UpcomingCampaigns.tsx

```
@@ -1,83 +1,321 @@
+    },
+    [expandedCampaignId, hasChanges],
+  );
+
+  // Using the original logic for displayedCampaigns
   const displayedCampaigns = expandedView ? campaigns : campaigns.slice(0, 3);
 
-  // Properly typed function to extract segment names
-  const getSegmentNames = (segments: Campaign['segments']) => {
-    if (!segments || !segments.included) return '';
+  const handleInputChange = useCallback(
+    (campaignId: number, field: keyof Campaign, value: any) => {
```

ESLint

[error] Unexpected any. Specify a different type.

(@typescript-eslint/no-explicit-any)